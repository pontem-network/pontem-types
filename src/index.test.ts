import { OverrideVersionedType } from '@polkadot/types/types/registry';
import { pontemDefinitions } from './index';

const fields: string[] = [
  'AccountInfo',
  'AuthorId',
  'Balance',
  'RegistrationInfo',
  'RoundIndex',
  'MoveModuleId',
  'MoveStructTag',
  'MoveTypeTag',
];

describe('Pontem Definitions', () => {
  test('Pontem Snapshot', () => {
    expect(pontemDefinitions).toMatchSnapshot();
  });

  test('Pontem Types', () => {
    expect(pontemDefinitions).toBeInstanceOf(Object);
    expect(pontemDefinitions).toHaveProperty('types');

    const types: OverrideVersionedType[] =
      pontemDefinitions.types as OverrideVersionedType[];

    expect(types).toBeInstanceOf(Array);
    expect(types.length).toBeGreaterThan(0);

    const type: OverrideVersionedType = types[0];
    expect(type).toHaveProperty('minmax');
    expect(type.minmax).toBeInstanceOf(Array);
    expect(type.minmax.length).toBe(2);

    expect(type).toHaveProperty('types');
    expect(type.types).toBeInstanceOf(Object);

    for (const field of fields) {
      expect(type.types).toHaveProperty(field);
      expect(type.types[field]).toBeTruthy();
    }
  });
});
