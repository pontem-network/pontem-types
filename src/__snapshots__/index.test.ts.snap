// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pontem Definitions Pontem Snapshot 1`] = `
Object {
  "rpc": Object {
    "mvm": Object {
      "estimateGasExecute": Object {
        "aliasSection": "mvm",
        "description": "estimate gas for script execution",
        "params": Array [
          Object {
            "name": "account",
            "type": "AccountId",
          },
          Object {
            "name": "tx_bc",
            "type": "Bytes",
          },
          Object {
            "name": "gas_limit",
            "type": "u64",
          },
        ],
        "type": "Estimation",
      },
      "estimateGasPublish": Object {
        "aliasSection": "mvm",
        "description": "estimate gas for module publishing",
        "params": Array [
          Object {
            "name": "account",
            "type": "AccountId",
          },
          Object {
            "name": "module_bc",
            "type": "Bytes",
          },
          Object {
            "name": "gas_limit",
            "type": "u64",
          },
        ],
        "type": "Estimation",
      },
      "gasToWeight": Object {
        "aliasSection": "mvm",
        "description": "convert gas to weight",
        "params": Array [
          Object {
            "name": "gas",
            "type": "u64",
          },
        ],
        "type": "Weight",
      },
      "weightToGas": Object {
        "aliasSection": "mvm",
        "description": "convert gas to weight",
        "params": Array [
          Object {
            "name": "weight",
            "type": "Weight",
          },
        ],
        "type": "u64",
      },
    },
  },
  "types": Array [
    Object {
      "minmax": Array [
        0,
        undefined,
      ],
      "types": Object {
        "AuthorId": "[u8;32]",
        "Balance": "u64",
        "Bond": Object {
          "amount": "Balance",
          "owner": "AccountId",
        },
        "Candidate": Object {
          "bond": "Balance",
          "fee": "Perbill",
          "id": "AccountId",
          "nominators": "Vec<Bond>",
          "state": "ValidatorStatus",
          "total": "Balance",
        },
        "Collator2": Object {
          "bond": "Balance",
          "bottom_nominators": "Vec<Bond>",
          "id": "AccountId",
          "nominators": "Vec<AccountId>",
          "state": "CollatorStatus",
          "top_nominators": "Vec<Bond>",
          "total_backing": "Balance",
          "total_counted": "Balance",
        },
        "CollatorSnapshot": Object {
          "bond": "Balance",
          "nominators": "Vec<Bond>",
          "total": "Balance",
        },
        "ExitQ": Object {
          "candidate_schedule": "Vec<(AccountId, RoundIndex)>",
          "candidates": "Vec<AccountId>",
          "nominator_schedule": "Vec<(AccountId, Option<AccountId>, RoundIndex)>",
          "nominators_leaving": "Vec<AccountId>",
        },
        "InflationInfo": Object {
          "annual": "RangePerbill",
          "expect": "RangeBalance",
          "round": "RangePerbill",
        },
        "MoveStructTag": Object {
          "address": "AccountId",
          "module": "Text",
          "name": "Text",
        },
        "MoveTypeTag": Object {
          "Address": "",
          "Bool": "",
          "Signer": "",
          "Struct": "MoveStructTag",
          "U128": "",
          "U64": "",
          "U8": "",
          "Vector": "MoveTypeTag",
          "_enum": Array [
            "Bool",
            "U8",
            "U64",
            "U128",
            "Address",
            "Signer",
            "Vector",
            "Struct",
          ],
        },
        "Nominator": Object {
          "nominations": "Vec<Bond>",
          "total": "Balance",
        },
        "Nominator2": Object {
          "nominations": "Vec<Bond>",
          "revocations": "Vec<AccountId>",
          "scheduled_revocations_count": "u32",
          "scheduled_revocations_total": "Balance",
          "status": "NominatorStatus",
          "total": "Balance",
        },
        "NominatorAdded": Object {
          "_enum": Object {
            "AddedToBottom": "Null",
            "AddedToTop": "Balance",
          },
        },
        "OrderedSet": "Vec<Bond>",
        "ParachainBondConfig": Object {
          "account": "AccountId",
          "percent": "Percent",
        },
        "Range": "RangeBalance",
        "RangeBalance": Object {
          "ideal": "Balance",
          "max": "Balance",
          "min": "Balance",
        },
        "RangePerbill": Object {
          "ideal": "Perbill",
          "max": "Perbill",
          "min": "Perbill",
        },
        "RegistrationInfo": Object {
          "account": "AccountId",
          "deposit": "Balance",
        },
        "RoundIndex": "u32",
        "RoundInfo": Object {
          "current": "RoundIndex",
          "first": "BlockNumber",
          "length": "u32",
        },
        "ValidatorStatus": Object {
          "_enum": Object {
            "Active": "Null",
            "Idle": "Null",
            "Leaving": "RoundIndex",
          },
        },
      },
    },
  ],
}
`;
